android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/permission/GeckoSitePermissionsStorage.kt

```diff
@@ -105,7 +105,7 @@ class GeckoSitePermissionsStorage(
 
         return onDiskStorage.getSitePermissionsPaged().map { onDiskPermission ->
             val geckoPermissions = geckoPermissionsByHost[onDiskPermission.origin].groupByType()
-            mergePermissions(onDiskPermission, geckoPermissions)
+            mergePermissions(onDiskPermission, geckoPermissions) ?: onDiskPermission
```

Comment:
@Amejia481 The alternative here was to use the not-null assertion operator. Is this mapping still correct?

Reply:
Could you help to clarify why do we need to provide a not-null value?

Reply:
The signature for the map method in `DataSource` class has changed in the new library version, and the code was migrated to Kotlin. It now expects a non-null result for mapping function, and the mergePermissions method returns nullable `SitePermissions`

Reply:
Yes, I think it's safe to add the Elvis operator!